packages/discovery-react-components/src/components/StructuredQuery/utils/stringifyStructuredQuerySelection.ts

```diff
@@ -0,0 +1,32 @@
+import { StructuredQuerySelection } from './structuredQueryInterfaces';
+
+export const stringifyStructuredQuerySelection = (
+  structuredQuerySelection: StructuredQuerySelection
+) => {
+  let stringifiedStructuredQuerySelection: string = '';
+  stringifiedStructuredQuerySelection += Object.keys(structuredQuerySelection.groups)
+    .map(groupId => {
+      return groupId === '0'
+        ? stringifyRows(structuredQuerySelection, groupId)
+        : '(' + stringifyRows(structuredQuerySelection, groupId) + ')';
+    })
+    .join(structuredQuerySelection.groups[0].operator);
+  return stringifiedStructuredQuerySelection;
+};
+
+const stringifyRows = (structuredQuerySelection: StructuredQuerySelection, groupId: string) => {
+  return structuredQuerySelection.groups[groupId].rows
+    .map((rowId: string) => {
+      const row = structuredQuerySelection.rows[rowId];
+      const rowField = containsReservedCharacters(row.field) ? '"' + row.field + '"' : row.field;
```

Comment:
i don't think we want to wrap field names in quotes as that query wouldn't actually work. is there a way we are testing the queries that this component produces?

Reply:
i believe if you want to escape the field names, you can use a backslash `\` and the query parser will still work

for example:
```
my,field:bar
```

would become
```
my\,field:bar
```

Reply:
Ah, okay, thanks for the catch and links with more info on this. Updated to escape the expanded set of reserved characters with a backslash 👍